src/BindPattern.jl

```diff
@@ -420,17 +440,25 @@ function bind_pattern!(
         pattern1 = shred_where_clause(guard, false, location, binder, assigned)
         pattern = BoundAndPattern(location, source, BoundPattern[pattern0, pattern1])
 
+    elseif is_expr(source, :call, 3) && source.args[1] == :(:)
+        # A range pattern.  We depend on the Range API to make sense of it.
```

Comment:
would we ever want to support the `start:step:stop` version as well? or is there not much use for it in practice?

Reply:
Yes, we should support that as well.  I'll update this.